homeassistant/components/forked_daapd/media_player.py

```diff
@@ -668,69 +668,68 @@ async def async_play_media(
         if media_type == MediaType.MUSIC:
             media_id = async_process_play_media_url(self.hass, media_id)
 
-            saved_state = self.state  # save play state
-            saved_mute = self.is_volume_muted
-            sleep_future = asyncio.create_task(
-                asyncio.sleep(self._tts_pause_time)
-            )  # start timing now, but not exact because of fd buffer + tts latency
-            await self._pause_and_wait_for_callback()
-            await self._save_and_set_tts_volumes()
-            # save position
-            saved_song_position = self._player["item_progress_ms"]
-            saved_queue = (
-                self._queue if self._queue["count"] > 0 else None
-            )  # stash queue
+            await self._async_announce(media_id)
```

Comment:
Just move the content to a separate function.
It is easier reviewed when hiding whitespaces.